v2/src/lib.rs

```diff
@@ -698,20 +709,76 @@ impl Storage for InfluxDbStorage {
         Ok(result)
     }
 
-    //putting a stub here to be implemented later
+    /// Gets the Latest of Each Key Expression from the database
+    /// Example:
+    /// Database contains the following (KeyExpr,Timestamp) tuples
+    /// [
+    ///     ("demo/a",t1),
+    ///     ("demo/a",t3),
+    ///     ("demo/b",t2),
+    ///     ("demo/b",t5),
+    ///     ("demo/a",t2),
+    /// ]
+    /// get_all_entries returns
+    /// [
+    ///     ("demo/a",t3),
+    ///     ("demo/b",t5)
+    /// ]
     async fn get_all_entries(&self) -> ZResult<Vec<(Option<OwnedKeyExpr>, Timestamp)>> {
-        tracing::warn!("!!! get_all_entries is NOT implemented yet !!!"); // See : https://github.com/ZettaScaleLabs/zenoh-backend-influxdb/pull/4
-        tracing::warn!("called get_all_entries in InfluxDBv2 storage");
-        return Ok(Vec::new());
-    }
-}
+        let mut result: Vec<(Option<OwnedKeyExpr>, Timestamp)> = Vec::new();
+
+        let db = self.get_db_name()?;
 
-fn get_db_name(config: StorageConfig) -> ZResult<String> {
-    let db = match config.volume_cfg.get(PROP_STORAGE_DB) {
-        Some(serde_json::Value::String(s)) => s.to_string(),
-        _ => bail!("no db was found"),
-    };
-    Ok(db)
+        let qs: String = format!(
+            "from(bucket: \"{}\")
+                    |> range(start:0)
+                    |> last()
+            ",
+            db
+        );
+
+        let query = Query::new(qs);
+
+        let vec_zpoint: Vec<ZenohPoint> = match self.client.query::<ZenohPoint>(Some(query)).await {
+            Ok(result) => result,
+            Err(e) => {
+                tracing::error!(
+                    "Couldn't get data from database {} in InfluxDBv2 storage with error: {} ",
+                    db,
+                    e
+                );
+                vec![]
```

Comment:
On error you return an empty vector instead of an actual error. Which means that from an application standpoint, an empty database or a faulty one will return the same: an empty vector. I suggest you return an error here.

Reply:
This makes sense.